hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/JoinPhysicalRule.java

```diff
@@ -103,7 +105,20 @@ public void onMatch(RelOptRuleCall call) {
                         logicalJoin.getJoinType()
                 );
                 call.transformTo(rel2);
+            } else {
+                call.transformTo(
+                        fail(logicalJoin, connector.typeName() + " connector doesn't support stream-to-batch JOIN")
+                );
             }
         }
     }
+
+    private ShouldNotExecuteRel fail(RelNode node, String message) {
+        return new ShouldNotExecuteRel(
+                node.getCluster(),
+                node.getTraitSet().replace(PHYSICAL),
+                node.getRowType(),
+                message
+        );
```

Comment:
Similar constructions are used in 10 places in the codebase, so we can have constructor `ShouldNotExecuteRel(RelNode node, String exceptionMessage)` and similarly for `MustNotExecutePhysicalRel`.